src/main/java/com/powsybl/openloadflow/sensi/DcSensitivityAnalysis.java

```diff
@@ -629,6 +629,7 @@ private void applyInjectionContingencies(Network network, LfNetwork lfNetwork, P
         for (LfGeneratorImpl generator : generators) {
             generator.setTargetP(0);
             if (distributedSlackOnGenerators && generator.isParticipating()) {
+                generator.setParticipating(false);
```

Comment:
This is never reverted, in state save/restore. You should add a test with more contingencies to highlight it.